indico/modules/events/papers/schemas.py

```diff
@@ -0,0 +1,126 @@
+from indico.modules.events.papers.models.revisions import PaperRevision, PaperRevisionState
+from indico.modules.users.schemas import UserSchema
+from indico.util.mimetypes import icon_from_mimetype
+
+
+class PaperEventSchema(mm.ModelSchema):
+    class Meta:
+        model = Event
+        fields = ('id', 'title', 'is_locked')
+
+
+class PaperFileSchema(mm.ModelSchema):
+    icon = Function(lambda paper_file: icon_from_mimetype(paper_file.content_type))
+
+    class Meta:
+        model = PaperFile
+        fields = ('id', 'revision_id', 'content_type', 'filename', 'icon')
+
+
+class PaperRevisionStateSchema(mm.Schema):
+    title = String()
+    css_class = String()
+
+
+class PaperAction(mm.Schema):
+    name = String()
+    css_class = String()
+
+
+class PaperCommentVisibilitySchema(mm.Schema):
+    name = String()
+    title = String()
+
+
+class PaperRevisionTimelineField(Field):
+    def _serialize(self, value, attr, obj, **kwargs):
+        if not value:
+            return []
+
+        serialized = []
+        for timeline_item in value:
+            serialized_item = {'timeline_item_type': timeline_item.timeline_item_type}
+            if timeline_item.timeline_item_type == 'comment':
+                serialized_item.update(paper_review_comment_schema.dump(timeline_item))
+            elif timeline_item.timeline_item_type == 'review':
+                serialized_item.update(paper_review_schema.dump(timeline_item))
+            serialized.append(serialized_item)
+        return serialized
+
+
+class PaperRevisionSchema(mm.ModelSchema):
+    submitter = Nested(UserSchema)
+    judge = Nested(UserSchema)
+    spotlight_file = Nested(PaperFileSchema)
+    files = List(Nested(PaperFileSchema))
+    state = EnumField(PaperRevisionState)
+    timeline = PaperRevisionTimelineField()
+    judgment_comment = Function(lambda revision: escape(revision.judgment_comment))
+
+    class Meta:
+        model = PaperRevision
+        fields = ('id', 'submitted_dt', 'judgment_dt', 'submitter', 'judge', 'spotlight_file', 'files',
+                  'is_last_revision', 'number', 'state', 'timeline', 'judgment_comment')
+
+
+class PaperReviewSchema(mm.ModelSchema):
+    score = Float(as_string=True)
+    created_dt = DateTime()
+    user = Nested(UserSchema)
+    modified_dt = DateTime()
+    visibility = Nested(PaperCommentVisibilitySchema)
+    proposed_action = Nested(PaperAction)
+    group = Function(lambda review: {'title': unicode(review.group.title)})
+    ratings = Function(lambda review: [{'value': rating.value} for rating in review.ratings])
+
+    class Meta:
+        model = PaperReview
+        fields = ('id', 'score', 'created_dt', 'user', 'comment', 'modified_dt', 'visibility', 'proposed_action',
+                  'group', 'ratings')
+
+
+class PaperReviewCommentSchema(mm.ModelSchema):
+    user = Nested(UserSchema)
+    text = String()
+    visibility = Nested(PaperCommentVisibilitySchema)
+    created_dt = DateTime()
```

Comment:
do you actually need this? it's a ModelSchema so most of the simple fields are usually autogenerated